unordered-containers.cabal

```
@@ -69,7 +77,15 @@ library
     MagicHash,
     BangPatterns
 
-  ghc-options: -Wall -O2 -fwarn-tabs -ferror-spans
+  if flag(noMsse42)
+    ghc-options: -Wall -O2 -fwarn-tabs -ferror-spans
+  else
+    -- only enable flag on x86
+    if arch(i386) || arch(x86_64)
+      ghc-options: -Wall -O2 -fwarn-tabs -ferror-spans -msse4.2
+    else
+    -- else, get whatever GHC gives you
+    ghc-options: -Wall -O2 -fwarn-tabs -ferror-spans
```

```suggestion
  ghc-options: -Wall -O2 -fwarn-tabs -ferror-spans
  if !flag(noMsse42) && arch(x86_64)
    -- only enable flag on x86
    ghc-options: -msse4.2
```
i386 doesn't support SSE4.2 (or SSE in general). There can be multiple `ghc-options` fields, so there's no need for this if-else chain.